### PR TITLE
[Unity][TVMScript] emit_te sugar

### PR DESCRIPTION
**Pull request by yongwww**
Add R.emit_te, a requested support https://github.com/tlc-pack/relax/issues/197

cc: @psrivas2 @Hzfengsy @YuchenJin @tqchen

### CODE REVIEW

**Review by tqchen — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @yongwww , as a followup, would be great if we can also confirm if it works on TVMScript parser

File: python/tvm/script/ir_builder/relax/ir.py [hunk 1]
```diff
@@ -325,6 +327,35 @@ def emit(value: Expr, annotate_struct_info: Optional[StructInfo] = None) -> Var:
...
+        to the PrimFunc that gets generated.
+
+    Returns
+    -------
+    var : Var
+        A newly created variable that gets bound to the call code.
+    """
+
+    # Levarage the util function call_te in Relax Block Blocker
+    emit_expr = rx_bb().call_te(func, *args, **kwargs)
```
Comment (psrivas2) [inline python/tvm/script/ir_builder/relax/ir.py:355]: this would emit the call binding correctly, but would it add the Primfunc to IRModule?

Comment (yongwww) [reply python/tvm/script/ir_builder/relax/ir.py:355]: Yes, the actual emit happens later. Actually the call_te is just a helper function, it is used to help prepare the expr to be emitted later.

**Comment by yongwww** [comment]
> Thanks @yongwww , as a followup, would be great if we can also confirm if it works on TVMScript parser

Thanks @tqchen! Is the following test case what you meant?

```python
@tvm.script.ir_module
    class Before:
        @R.function
        def main(x: R.Tensor((10, 20), "float32")) -> R.Tensor(None, "float32", ndim=2):
            # block 0
            with R.dataflow():
                lv0 = R.emit_te(topi.add, x, relax.const(1, "float32"))
                lv1 = R.emit_te(topi.exp, lv0)
                gv = R.emit_te(topi.squeeze, lv1)
                R.output(gv)
            return gv

    Before.show()
```

output of Before.show():
```python
@I.ir_module
class Module:
    @R.function
    def main(
        x: R.Tensor((10, 20), dtype="float32")
    ) -> R.Tensor((10, 20), dtype="float32"):
        with R.dataflow():
            lv = R.call_tir(
                add,
                (x, R.const(1, "float32")),
                out_sinfo=R.Tensor((10, 20), dtype="float32"),
            )
            lv0: R.Tensor((10, 20), dtype="float32") = lv
            lv2 = R.call_tir(exp, (lv0,), out_sinfo=R.Tensor((10, 20), dtype="float32"))
            lv1: R.Tensor((10, 20), dtype="float32") = lv2
            lv4 = R.call_tir(
                squeeze, (lv1,), out_sinfo=R.Tensor((10, 20), dtype="float32")
            )
            gv: R.Tensor((10, 20), dtype="float32") = lv4
            R.output(gv)
        return gv
```

**Comment by tqchen** [comment]
Ah sorry i missed it!

**Comment by psrivas2** [comment]
Thanks for example @yongwww . In the output of `Before.show()` PrimFuncs for `add`, `squeeze` and `exp` are not emitted. I only see the Relax function `main` present in the module.

**Comment by yongwww** [comment]
> Thanks for example @yongwww . In the output of `Before.show()` PrimFuncs for `add`, `squeeze` and `exp` are not emitted. I only see the Relax function `main` present in the module.

ohhhhhh, i see, thanks for pointing this out!!! Will fix it by end of week.

**Pull request merged**